src/main/java/com/clouway/push/client/ChannelApiPushEventBus.java

```diff
@@ -40,20 +40,15 @@ public HandlerRegistration addHandler(final PushEvent.Type type, final PushEvent
 
     final HandlerRegistration[] handlerRegistration = {null};
 
-    if (!pushChannelApi.hasOpenedChannel()) {
+    if (!pushChannelApi.hasInitialConnection()) {
 
-      pushChannelApi.connect(new AsyncConnectCallback() {
+      pushChannelApi.connect();
```

Comment:
You are absolutely right. We could add some kind of ConnectionStatusListener, to provide events when connection is opened, when connection is retrying and etc. So we will be able to handle this in app that is using this library to indicate properly what is happening with the push notification.